internal/cmd/connect/command_plugin_install.go

```diff
@@ -0,0 +1,456 @@
+		if ins == nil {
+			ins, err = getConfluentPlatformInstallation(cmd, force)
+			if err != nil {
+				return err
+			}
+		}
+		workerConfigs, err = chooseWorkerConfigs(cmd, ins, force)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(workerConfigs) > 0 {
+		updateWorkerMsg := "\nAdding plugin installation directory to the plugin path in the following files:"
+		for _, workerConfig := range workerConfigs {
+			updateWorkerMsg = fmt.Sprintf("%s\n\t* %v", updateWorkerMsg, workerConfig)
+		}
+		output.Println(updateWorkerMsg)
+	}
+
+	for _, workerConfig := range workerConfigs {
+		if err := updateWorkerConfig(pluginDir, workerConfig, dryRun); err != nil {
+			return err
+		}
+	}
+
+	if dryRun {
+		output.Println("\n[DRY RUN] Dry run successfully completed. All requirements are met; you may proceed with the installation.")
+	}
+
+	return nil
+}
+
+func parsePluginId(plugin string) (string, string, string, error) {
+	parsePluginErrorMsg := "plugin not found"
+	parsePluginSuggestions := "Provide either a path to a local file or a plugin id from Confluent Hub with the format: `<owner>/<name>:<version>`."
+
+	ownerNameSplit := strings.Split(plugin, "/")
+	if len(ownerNameSplit) != 2 || ownerNameSplit[0] == "" {
+		return "", "", "", errors.NewErrorWithSuggestions(parsePluginErrorMsg, parsePluginSuggestions)
+	}
+	nameVersionSplit := strings.Split(ownerNameSplit[1], ":")
+	if len(nameVersionSplit) != 2 || nameVersionSplit[0] == "" || nameVersionSplit[1] == "" {
+		return "", "", "", errors.NewErrorWithSuggestions(parsePluginErrorMsg, parsePluginSuggestions)
+	}
+
+	return ownerNameSplit[0], nameVersionSplit[0], nameVersionSplit[1], nil
+}
+
+func getLocalManifest(archivePath string) (*manifest, error) {
+	zipReader, err := zip.OpenReader(archivePath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to open local archive file %s", archivePath)
+	}
+	defer zipReader.Close()
+
+	for _, zipFile := range zipReader.File {
+		isManifest, err := filepath.Match("*/manifest.json", filepath.ToSlash(zipFile.Name))
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to examine file %s inside local archive file %s", zipFile.Name, archivePath)
+		}
+
+		if isManifest {
+			manifestFile, err := zipFile.Open()
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to open manifest file %s inside local archive file %s", zipFile.Name, archivePath)
+			}
+			defer manifestFile.Close()
+
+			jsonByteArr, err := io.ReadAll(manifestFile)
+			if err != nil {
+				return nil, err
+			}
+
+			pluginManifest := new(manifest)
+			if err := json.Unmarshal(jsonByteArr, &pluginManifest); err != nil {
+				return nil, err
+			}
+
+			return pluginManifest, nil
+		}
+	}
+
+	return nil, errors.Errorf(`failed to find manifest file inside local archive file "%s"`, archivePath)
+}
+
+func getRemoteManifest(owner, name, version string) (*manifest, error) {
+	manifestUrl := fmt.Sprintf("https://api.hub.confluent.io/api/plugins/%s/%s", owner, name)
```

Comment:
Is there a Confluent Hub SDK? Would it be overkill to create one? It might be useful if we add more functionality in the future (i.e. searching for plugins). At the very least, this should probably be moved to a separate file or package.

Reply:
I don't think there is. I wasn't even able to find documentation on this API at all. I'll move it somewhere though.